addons/website_slides/models/slide_slide.py

```diff
@@ -356,6 +357,29 @@ def _notify_get_groups(self, message, groups):
 
         return groups
 
+    # ---------------------------------------------------------
+    # Access Rights Methods
+    # ---------------------------------------------------------
+
+    def _check_read_access(self):
+        try:
+            self.check_access_rights('read')
+            self.check_access_rule('read')
+        except AccessError:
+            return False
+        return True
+
+    def _get_slide_action_access(self):
+        result = dict()
+        for slide in self:
+            result.setdefault(slide.id, dict(can_access=False, can_vote=False, can_comment=False))
+            can_access = slide._check_read_access()
+            if can_access and (slide.channel_id.is_member or slide.is_preview or self.env.user.has_group('website.group_website_publisher')):
```

Comment:
`or slide.can_publish` instead of `self.env.user.has_group('website.group_website_publisher'))`